### PR TITLE
feat(My Nintendo): Complete recode

### PR DESCRIPTION
**Pull request by ACertainCoder**
**Describe the changes in this pull request:**
Completely recoded the My Nintendo presence.

<details>
<summary> Proof to proving the creation/modification working </summary>
<br>
My Nintendo

</details>

### CODE REVIEW

File: websites/M/My Nintendo/presence.ts [hunk 1]
```diff
@@ -7,138 +7,157 @@ const presence = new Presence({
   });
```
Comment (Bas950) [inline websites/M/My Nintendo/presence.ts:7]: use general strings

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
- Why did you decide to recode this? 
- Why don't you use a Record instead of the extensive switch-case statement?
- How is the multiLang change handled?

Please remove the exessive:
- Obvious comments, keep them where appropriate, not everywhere 
- `presence.setActivity()`s

File: websites/M/My Nintendo/presence.ts [hunk 2]
```diff
@@ -1,144 +1,182 @@
+/**
+ * Language Strings
+ */
+interface LangStrings {
+  browsing: string;
+  reading: string;
+}
```
Comment (EncryptedDev) [inline websites/M/My Nintendo/presence.ts:7]:
```suggestion
```

File: websites/M/My Nintendo/presence.ts [hunk 3]
```diff
@@ -1,144 +1,182 @@
...
+      presence.setActivity();
+      break;
+  }
+});
+
+/**
+ * Get Language Strings
+ * @returns Language Strings
+ */
+async function getStrings(): Promise<LangStrings> {
```
Comment (EncryptedDev) [inline websites/M/My Nintendo/presence.ts:62]:
```suggestion
async function getStrings() {
```

File: websites/M/My Nintendo/presence.ts [hunk 4]
```diff
@@ -1,144 +1,182 @@
...
+  }
+
+  presence.setActivity(presenceData);
+}
+
+/**
+ * Handle points page
+ * @param path URL path
+ */
+async function handlePoints(path: string[]): Promise<void> {
```
Comment (EncryptedDev) [inline websites/M/My Nintendo/presence.ts:122]: Why is this async?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/M/My Nintendo/presence.ts (hunk 4 above)
Comment (ACertainCoder) [reply websites/M/My Nintendo/presence.ts:111]: Consistency

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/M/My Nintendo/presence.ts [hunk 5]
```diff
@@ -1,144 +1,117 @@
...
-      //Rewards
-      case "rewards":
-        presenceData.smallImageKey = "reading";
-        presenceData.smallImageText = (await strings).reading;
+    presenceData: PresenceData = {};
 
-        if (!path.includes("media")) {
-          presenceData.details =
-            document.getElementsByClassName("PageHeader_title")[0].textContent;
+  strings = await getStrings();
```
Comment (Bas950) [inline websites/M/My Nintendo/presence.ts:11]: only fetch on start of when changed

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/M/My Nintendo/presence.ts [hunk 6]
```diff
@@ -1,144 +1,117 @@
...
+          ".PageSubHeader_title"
+        )?.textContent;
+      }
+      break;
+    // Redeem Point Codes
+    case "serial_number":
+      presenceData.details =
+        document.querySelector<HTMLHeadingElement>("h1")?.textContent ??
+        document.title;
+      presenceData.largeImageKey = "logo_big";
```
Comment (EncryptedDev) [inline websites/M/My Nintendo/presence.ts:89]: Set this in the PresenceData object on L9

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**